dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/liberty/CDIRuntimeImpl.java

```diff
@@ -225,6 +225,10 @@ protected void setExtensionService(ServiceReference<WebSphereCDIExtension> refer
 
     protected void unsetExtensionService(ServiceReference<WebSphereCDIExtension> reference) {
         extensionsSR.removeReference(reference);
+        CDIContainerImpl cdiContainer = getCDIContainer();
+        if (cdiContainer != null) {
+            cdiContainer.removeRuntimeExtensionArchive(reference);
```

Comment:
Might also want to add a comment here to explain why we have to do this.

Reply:
done